test/manipulation.test.js

```diff
@@ -91,6 +118,20 @@ describe('manipulation', function () {
       });
     });
 
+    it('should not allow user-defined value for the id of object - create (promise variant)', function (done) {
+      Person.create({id: 123456})
+      .then (function (p) {
+        should.not.exist(p);
+        done();
+      })
+      .catch(function (err) {
+        err.should.be.instanceof(ValidationError);
+        err.statusCode.should.equal(422);
+        err.details.messages.id.should.eql(['can\'t be set']);
+        done();
+      });
```

Comment:
> In this case, your second catch statement wouldn't do anything unless
another error is thrown from the first error handler, which it isn't.

The second catch handler is invoked when the assert in the first *success handler* throws an `AssertionError`.